app/forms/optional_consent_form.rb

```diff
@@ -0,0 +1,28 @@
+class OptionalConsentForm < QuestionsForm
+  set_attributes_for(
+    :consent,
+    :ip,
+    :disclose_consented,
+    :global_carryforward_consented,
+    :relational_efin_consented,
+    :use_consented,
+    :user_agent
+  )
+
+  def save
+    attributes = attributes_for(:consent)
+
+    current_time = DateTime.now
+    consent_attrs = {
+      disclose_consented_at: attributes[:disclose_consented] == "true" ? current_time : nil,
+      global_carryforward_consented_at: attributes[:global_carryforward_consented] == "true" ? current_time : nil,
+      relational_efin_consented_at: attributes[:relational_efin_consented] == "true" ? current_time : nil,
+      use_consented_at: attributes[:use_consented] == "true" ? current_time : nil,
+      user_agent: attributes[:user_agent],
+      ip: attributes[:ip]
+    }
+    intake.client.consent.present? ?
```

Comment:
ohh smart.

Reply:
You could also use find_or_initialize by here which _might_ save one RT to the db? Because I think you'll have to do a query for the consent to see if its present, then another to update or create. But I guess with find or initialize by, you'd just be grabbing the record and then updating or using the same record so ... actually this is probably simpler!